src/deadline/job_attachments/asset_sync.py

```diff
@@ -571,6 +571,10 @@ def sync_outputs(
                     local_root = Path(
                         storage_profiles_path_mapping_rules[manifest_properties.rootPath]
                     )
+                    # We use this session_dir to filter out any files resolved to a location
+                    # outside of that directory. If storage profile path is available, we can
+                    # consider the session directory to be the storage profile path.
+                    session_dir = local_root
```

Comment:
Does this cover cases where the job simultaneously has the following?

1. A manifest that is for file system location A in the storage profile.
2. Another manifest that is for file system location B in the storage profile.
3. A third manifest that is not for a file system location, so appears in the session directory.

Reply:
We iterate over each manifest here: https://github.com/aws-deadline/deadline-cloud/blob/f67d0a1458879dac8f253b5445f599cbb0912fed/src/deadline/job_attachments/asset_sync.py#L564

Each manifest object here has it's own associated root path that may or may not map to a storage profile location: https://github.com/aws-deadline/deadline-cloud/blob/f67d0a1458879dac8f253b5445f599cbb0912fed/src/deadline/job_attachments/models.py#L162

So I believe this covers the cases you're talking about, as each manifest would be grouped under a different root path

Reply:
Oh I see what you mean though, since we're overwriting the session_dir, we would have it overwritten for the next iteration of the loop. Yea we should just make a local variable for the session_dir

Reply:
Thank you. I added a variable `session_root` local to that loop instead of overwriting the `session_dir`:
```
for manifest in manifests:
    session_root = session_dir
    ...
    if local storage profile is available:
      session_root = local_root
    ...
    _get_output_files(..., local_root, session_root)
```